### PR TITLE
feat: support payloads bigger than the MaxSendStreamSize

### PR DESCRIPTION
**Pull request by richardapeters**

### CODE REVIEW

File: services/network/Http.hpp [hunk 1]
```diff
@@ -125,6 +126,7 @@ namespace services
         HttpHeader hostHeader;
         const HttpHeaders headers;
         bool chunked = false;
+        bool sentHeader = false;
```
Comment (daantimmer) [inline services/network/Http.hpp:129]:
```suggestion
        bool chunked{false};
        bool sentHeader{false};
```

File: services/network/HttpClientImpl.cpp [hunk 2]
```diff
@@ -360,11 +360,21 @@ namespace services
     void HttpClientImpl::SendingStateRequest::SendStreamAvailable(infra::SharedPtr<infra::StreamWriter>&& writer)
     {
         infra::TextOutputStream::WithErrorPolicy stream(*writer);
-        client.request->Write(stream);
-        client.request = infra::none;
-        writer = nullptr;
+        auto amount = client.request->Write(stream);
+        client.request->Consume(amount);
```
Comment (daantimmer) [inline services/network/HttpClientImpl.cpp:364]:
```suggestion
        client.request->Consume(client.request->Write(stream));
```

File: services/network/Http.cpp [hunk 3]
```diff
@@ -100,25 +100,40 @@ namespace services
...
+    std::size_t HttpRequestFormatter::Write(infra::TextOutputStream stream) const
     {
-        stream << verb << sp << requestTarget << sp << httpVersion << crlf;
+        if (!sentHeader)
+        {
+            stream << verb << sp << requestTarget << sp << httpVersion << crlf;
 
-        for (auto&& header : headers)
-            stream << header << crlf;
+            for (auto&& header : headers)
```
Comment (daantimmer) [inline services/network/Http.cpp:115]:
```suggestion
            for (const auto& header : headers)
```

File: services/network/test_doubles/ConnectionStub.hpp [hunk 4]
```diff
@@ -30,6 +30,7 @@ namespace services
         void SimulateDataReceived(infra::ConstByteRange data);
 
         std::vector<uint8_t> sentData;
+        std::size_t maxSendStreamSize = 1024;
```
Comment (daantimmer) [inline services/network/test_doubles/ConnectionStub.hpp:33]:
```suggestion
        const std::size_t maxSendStreamSize{ 1024 };
```

Comment (richardapeters) [reply services/network/test_doubles/ConnectionStub.hpp:33]: The intention was that tests can change this parameter

**Pull request merged**